x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/TimeseriesLifecycleType.java

```diff
@@ -473,6 +474,109 @@ static void validateDownsamplingIntervals(Collection<Phase> phases) {
+        if (firstReplicateFor != null) {
+            final Map<String, Phase> allPhases = phases.stream().collect(Collectors.toMap(Phase::getName, Function.identity()));
+            final List<Phase> allPhasesInOrder = INSTANCE.getOrderedPhases(allPhases);
+
+            // find the 'implied min_age' of the phase that contains the searchable_snapshot action with a replicate_for,
+            // it's the latest non-zero min_age of the phases up to and including the phase in question (reminder that min_age values
+            // are either absent/zero or increasing)
+            TimeValue impliedMinAge = TimeValue.ZERO;
+            for (Phase phase : allPhasesInOrder) {
+                // if there's a rollover (in the hot phase) then the hot phase is implicitly a 'zero', since we calculate subsequent
+                // phases from the time of *rollover*
+                final var phaseMinAge = phase.getActions().containsKey(RolloverAction.NAME) ? TimeValue.ZERO : phase.getMinimumAge();
+
+                // TimeValue.ZERO is the null value for minimumAge in Phase
+                if (phaseMinAge != TimeValue.ZERO) {
+                    impliedMinAge = phaseMinAge;
+                }
+                // loop until we find the phase that has the searchable_snapshot action with a replicate_for
+                if (phase.getName().equals(firstSearchableSnapshotPhase)) {
+                    break;
+                }
+            }
+
+            boolean afterReplicatorFor = false;
+            for (Phase phase : allPhasesInOrder) {
+                // loop until we find the phase after the one that has a searchable_snapshot with replicate_for
+                if (phase.getName().equals(firstSearchableSnapshotPhase)) {
+                    afterReplicatorFor = true;
+                    continue; // because we don't want to check the min_age on *this* phase, but on the next ones
+                }
+                // check the min_age requirement for all phases after the one that has the replicate_for set
+                if (afterReplicatorFor) {
+                    final var phaseMinAge = phase.getMinimumAge();
+                    // TimeValue.ZERO is the null value for minimumAge in Phase
+                    final long minAgeDeltaMillis = phaseMinAge.millis() - impliedMinAge.millis();
+                    if (phaseMinAge != TimeValue.ZERO && minAgeDeltaMillis < firstReplicateFor.millis()) {
+                        throw new IllegalArgumentException(
+                            Strings.format(
+                                "the value for replicate_for [%s] must be less than or equal to the difference in min_age for "
+                                    + "the current phase and subsequent phases, but the [%s] phase has a min_age of [%s] "
+                                    + "and the [%s] phase has a min_age of [%s], so the difference was only [%s]",
```

Comment:
I think we can make this clearer to people that aren't familiar with the technical wording, perhaps something like:

"The time a searchable snapshot is replicated in replicate_for […] may not exceed the time the next phase is configured to begin. Based on the min_age of […] for the […] phase, the maximum time the snapshot could be replicated is […]."

What do you think?

Reply:
I went with your message (approximately!) in 39e6660644a, but I reworded it very slightly -- feel free to insist on different wording than what I happened to go with.